prow/plugins/approve/approvers/owners.go

```diff
@@ -428,19 +429,21 @@ func (ap Approvers) UnapprovedFiles() sets.String {
 }
 
 // GetFiles returns owners files that still need approval.
-func (ap Approvers) GetFiles(org, repo, branch string) []File {
+func (ap Approvers) GetFiles(linkURL *url.URL, org, repo, branch string) []File {
```

Comment:
Wait, am I reading this right that `GetFiles` is only called in test? Where else is this code used?

Reply:
`GetFiles` is called by `GenerateTemplate` inside `GetMessage` to build `[APPROVALNOTIFIER]` github comments, here: https://github.com/kubernetes/test-infra/blob/d3ac65fa12d9f2579350772ee8e4e8f956ed9652/prow/plugins/approve/approvers/owners.go#L649

Reply:
Wait so why can't we pass the link URL into that func instead of embedding it into every single fiel?

Reply:
Having it as a field on the files means it can change between files which isnt true

Reply:
I agree, but I'd we follow that logic train a little further, we realize that org and repo won't change on a file basis either, so those should be removed also. This implementation is as close to the original implementation that I can think of, so the risk of unexpected side effects should be small.

Can we create an enhancement issue to rework this and keep this change smaller?

I also don't like that the template logic is split in multiple places, that is not ideal.